### PR TITLE
Tp test doc

### PR DESCRIPTION
**Pull request by nick-drozd**
Aside from adding documentation, this PR moderately rewrites the XO TP test to match the documentation and also renames some other parts of the TP test infrastructure (but without any functional changes).

### CODE REVIEW

File: docs/source/app_developers_guide/testing.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+*******************************
+
+Say we've written a transaction processor -- how can it be tested? In this section, we'll see how to test the XO transaction processor described in :doc:`txn_family_tutorial` against a mock validator.
+
+Communication with the Validator
+================================
+
+First, let's look at what happens when the XO transaction 'create game000' is processed. Communication between the validator and a transaction processor (TP) takes place in six steps:
+
+1) The validator needs to process a transaction, so it sends a transaction process request to the TP containing the transaction information (in this case, the command 'create' and the game name 'game000'.
```
Comment (aludvik) [inline docs/source/app_developers_guide/testing.rst]: Missing ")"

File: docs/source/app_developers_guide/testing.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+2) The TP needs to know whether there already exists a game with the name 'game000', so it sends a 'get' request back to the validator.
```
Comment (aludvik) [inline docs/source/app_developers_guide/testing.rst]: -> "whether a game already exists with the name"

File: docs/source/app_developers_guide/testing.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+3) After querying state, the validator sends back the results to the TP (in this case, we'll assume that there is no game called 'game000').
+
+4) The TP now sends a message back to the validator requesting that it set 'game000' to an initialized game (with an empty board, etc).
+
+5) The validator sends the address of the created game back to the TP.
+
+6) The TP sends back an 'OK' response.
```
Comment (aludvik) [inline docs/source/app_developers_guide/testing.rst]: The TP sends an 'OK' response to the validator, signaling that the transaction was processed successfully.

File: docs/source/app_developers_guide/testing.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        self.validator_responds_to_set_request(set_message, 'game000')
+        
+        self.validator_expects_tp_response('OK')
+
+Message Factory and Mock Validator
+==================================
+
+To implement these functions, we'll need a mock validator to send messages to and from and a way manufacture the appropriate messages. Manufacturing messages is not terribly interesting -- it's just a matter of getting the relevant information into probuf objects. This will be taken care of for us by the *XoMessageFactory* (``/project/sawtooth-core/sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py``), which itself relies on the more general *MessageFactory* (``/project/sawtooth-core/sdk/python/sawtooth_processor_test/message_factory.py``).
+
+To set up a mock validator, we can rely on *TransactionProcessorTestCase* (``/project/sawtooth-core/sdk/python/sawtooth_processor_test/transaction_processor_test_case.py``). This test class assumes that a transaction processor has been set up with a certain address, and creates a mock validator to connect at that address (and closing that connection after all the test have run):
```
Comment (aludvik) [inline docs/source/app_developers_guide/testing.rst]: This test class assumes that a transaction processor has been set up to connect to a certain address. It creates a mock validator that binds to that address (and closes the connection after all the tests have run):

File: docs/source/app_developers_guide/testing.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            state='P1-NEXT', player1='', player2='')
+        
+        self.validator_responds_to_set_request(set_message, 'game000')
+        
+        self.validator_expects_tp_response('OK')
+
+Message Factory and Mock Validator
+==================================
+
+To implement these functions, we'll need a mock validator to send messages to and from and a way manufacture the appropriate messages. Manufacturing messages is not terribly interesting -- it's just a matter of getting the relevant information into probuf objects. This will be taken care of for us by the *XoMessageFactory* (``/project/sawtooth-core/sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py``), which itself relies on the more general *MessageFactory* (``/project/sawtooth-core/sdk/python/sawtooth_processor_test/message_factory.py``).
```
Comment (nick-drozd) [inline docs/source/app_developers_guide/testing.rst]: Actually I would say that the protobuf stuff is very involved, which is part of the reason I find it so uninteresting! But I take your point, and I'll make it more neutral.

**Pull request merged**